From 403c56b92ff5db28385bf0f9bd6dc1a80782293f Mon Sep 17 00:00:00 2001
From: Huang Chaofan <chaofan.huang@nxp.com>
Date: Thu, 29 Nov 2018 19:31:18 +0800
Subject: [PATCH 5241/5242] MLK-20494: Adjust update stream function for the
 format that need add start code

commit  66620c3d281c5a5b27cbb7a51276d2f94f619f1c from
https://source.codeaurora.org/external/imx/linux-imx.git

Adjust update stream function for the format that need add start code

Signed-off-by: Huang Chaofan <chaofan.huang@nxp.com>
(cherry picked from commit 3b29245bc201f1090e46c70ca60e61093e27675e)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/mxc/vpu-decoder-b0/vpu_b0.c |    6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/drivers/mxc/vpu-decoder-b0/vpu_b0.c b/drivers/mxc/vpu-decoder-b0/vpu_b0.c
index ec10bdc..bb11695 100644
--- a/drivers/mxc/vpu-decoder-b0/vpu_b0.c
+++ b/drivers/mxc/vpu-decoder-b0/vpu_b0.c
@@ -1549,7 +1549,7 @@ static int update_stream_addr(struct vpu_ctx *ctx, void *input_buffer, uint32_t
 	uint32_t index = ctx->str_index;
 	pSTREAM_BUFFER_DESCRIPTOR_TYPE pStrBufDesc;
 	struct queue_data *q_data = &ctx->q_data[V4L2_SRC];
-	u_int8 payload_header[256];
+	u_int8 payload_header[256] = {0};
 	uint32_t nfreespace = 0;
 	uint32_t wptr;
 	uint32_t rptr;
@@ -1632,6 +1632,9 @@ static int update_stream_addr(struct vpu_ctx *ctx, void *input_buffer, uint32_t
 				wptr = start + buffer_size - (end-wptr);
 			}
 		} else {
+			memcpy(wptr_virt, payload_header, length);
+			wptr += length;
+			wptr_virt += length;
 			memcpy(wptr_virt, input_buffer, buffer_size);
 			wptr += buffer_size;
 		}
@@ -3084,6 +3087,7 @@ static int v4l2_open(struct file *filp)
 	atomic64_sub(sizeof(MediaIPFW_Video_SeqInfo), &ctx->statistic.total_alloc_size);
 	release_queue_data(ctx);
 err_alloc_seq:
+	remove_instance_file(ctx);
 	kfifo_free(&ctx->msg_fifo);
 err_alloc_fifo:
 	destroy_workqueue(ctx->instance_wq);
-- 
1.7.9.5

